pkg/controller/istiocontrolplane/istiocontrolplane_controller.go

```diff
@@ -44,13 +45,17 @@ const finalizer = "istio-operator"
 // Add creates a new IstioControlPlane Controller and adds it to the Manager. The Manager will set fields on the Controller
 // and Start it when the Manager is Started.
 func Add(mgr manager.Manager) error {
-	return add(mgr, newReconciler(mgr))
+	return add(mgr, newReconciler(mgr, v1))
 }
 
 // newReconciler returns a new reconcile.Reconciler
-func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+func newReconciler(mgr manager.Manager, apiVersion string) reconcile.Reconciler {
```

Comment:
Why not just have newReconcilerV1 and newReconcilerV2?

Reply:
updated to have v1 and v2